### PR TITLE
docs: fix & improve parallel algorithms documentation 4

### PR DESCRIPTION
**Pull request by bhumitattarde**

### CODE REVIEW

File: libs/core/algorithms/include/hpx/parallel/algorithms/sort_by_key.hpp [hunk 1]
```diff
@@ -47,15 +47,16 @@ namespace hpx { namespace parallel { inline namespace v1 {
...
     ///                     comparisons.
     ///
-    /// A sequence is sorted with respect to a comparator \a comp and a
-    /// projection \a proj if for every iterator i pointing to the sequence and
+    /// A sequence is sorted with respect to a comparator \a comp
+    /// if for every iterator i pointing to the sequence and
     /// every non-negative integer n such that i + n is a valid iterator
     /// pointing to an element of the sequence, and
+    /// FIXME(bhumit,hkaiser): proj doesn't exist
     /// INVOKE(comp, INVOKE(proj, *(i + n)), INVOKE(proj, *i)) == false.
```
Comment (bhumitattarde) [inline libs/core/algorithms/include/hpx/parallel/algorithms/sort_by_key.hpp:60]: @hkaiser what should this be now that `proj` doesn't exist?

Comment (hkaiser) [reply libs/core/algorithms/include/hpx/parallel/algorithms/sort_by_key.hpp:60]: `INVOKE(comp, *(i + n), *i) == false`

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**